yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -679,6 +755,9 @@ object Client extends Logging {
   // Distribution-defined classpath to add to processes
   val ENV_DIST_CLASSPATH = "SPARK_DIST_CLASSPATH"
 
+  // Subdirectory where the user's hadoop config files are written in the app staging dir.
+  val HADOOP_CONF_DIR = "__hadoop_conf__"
```

Comment:
HADOOP_CONF_DIR is the name hadoop uses for the env variable.  To avoid confusion with that could we rename this variable?

Also (minor nit) you might want to remove the "in the app staging dir" part of the comment.  Perhaps replace with when localized.  When its in the app staging dir its call __hadoop_conf__.zip, then it gets localized then it would go into the directory.